addons/l10n_it_edi/models/account_move.py

```diff
@@ -518,42 +535,54 @@ def _l10n_it_edi_document_type_mapping(self):
                      'simplified': False,
                      'self_invoice': True,
                      'partner_country_code': "SM"},
+            'TD16': {'move_types': ['in_invoice', 'in_refund'],
+                     'import_type': 'in_invoice',
+                     'simplified': False,
+                     'self_invoice': True,
+                     'tax_tags': {'VJ6', 'VJ7', 'VJ8', 'VJ12', 'VJ13', 'VJ14', 'VJ15', 'VJ16', 'VJ17'}},
             'TD17': {'move_types': ['in_invoice', 'in_refund'],
                      'import_type': 'in_invoice',
                      'simplified': False,
                      'self_invoice': True,
-                     'services_or_goods': "service"},
+                     'services_or_goods': "service",
+                     'tax_tags': {'VJ3'}},
             'TD18': {'move_types': ['in_invoice', 'in_refund'],
                      'import_type': 'in_invoice',
                      'simplified': False,
                      'self_invoice': True,
                      'services_or_goods': "consu",
                      'goods_in_italy': False,
-                     'partner_in_eu': True},
+                     'partner_in_eu': True,
+                     'tax_tags': {'VJ9'}},
             'TD19': {'move_types': ['in_invoice', 'in_refund'],
                      'import_type': 'in_invoice',
                      'simplified': False,
                      'self_invoice': True,
                      'services_or_goods': "consu",
-                     'goods_in_italy': True},
+                     'goods_in_italy': True,
+                     'tax_tags': {'VJ3'}},
         }
 
     def _l10n_it_edi_get_document_type(self):
-        """ Compare the features of the invoice to the requirements of each TDxx FatturaPA
-            document type until you find a valid one. """
+        """ Compare the features of the invoice to the requirements of each Document Type (TDxx)
+        FatturaPA until you find a valid one. """
+
+        def compare(actual_values, expected_values):
+            """ Compare a single entry from the invoice features with the one of the document_type """
+            if isinstance(expected_values, set | list | tuple):
+                # i.e. When we compare actual tax_tags from the invoice with expected tags, we see if there is at least one in common
+                if isinstance(actual_values, set):
+                    return actual_values & set(expected_values)
+                # i.e. When we compare the move_type with the available ones, these can be more than one
+                return actual_values in expected_values
```

Comment:
I still wonder if this is not equivalent to:
set([actual_values]) in set(expected_values)

Reply:
This seems ok: actual_values is also single if it is not a set.